src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectSnapshotChangeTrigger.cs

```diff
@@ -72,6 +74,25 @@ internal void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs
                         break;
                     }
 
+                case WorkspaceChangeKind.DocumentChanged:
+                case WorkspaceChangeKind.DocumentReloaded:
+                    {
+                        // This is the case when a component declaration file changes on disk. We have an MSBuild
+                        // generator configured by the SDK that will poke these files on disk when a component
+                        // is saved, or loses focus in the editor.
+                        project = e.OldSolution.GetProject(e.ProjectId);
+                        var document = project.GetDocument(e.DocumentId);
+                        
+                        // Using EndsWith because Path.GetExtension will ignore everything before .cs
+                        // Using Ordinal because the SDK generates these filenames.
+                        if (document.FilePath != null && document.FilePath.EndsWith(".cshtml.g.cs", StringComparison.Ordinal))
```

Comment:
Is there some sort of FileKind static we should be using here? Feels odd just having this out in the blue. Also, components get generated as `.cshtml.g.cs`? Not `.razor.g.cs`?

Reply:
This is defined in msbuild so no. For preview2 we only support `.cshtml` because the partner work hasn't even started.